src/zenml/integrations/openai/hooks/open_ai_failure_hook.py

```diff
@@ -0,0 +1,83 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Functionality for standard hooks."""
+
+import io
+import sys
+from typing import TYPE_CHECKING
+
+import openai
+from rich.console import Console
+
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
+if TYPE_CHECKING:
+    from zenml.steps import BaseParameters, StepContext
+
+
+def openai_alerter_failure_hook(
+    context: "StepContext", params: "BaseParameters", exception: BaseException
+) -> None:
+    """Failure hook that executes after step fails, with OpenAI GPT support.
+
+    This hook uses OpenAI's LLM to generate suggestions on how to fix a step
+    that fails.
+
+    Args:
+        context: Context of the step.
+        params: Parameters used in the step.
+        exception: Original exception that lead to step failing.
+    """
+    if context.stack and context.stack.alerter:
+        output_captured = io.StringIO()
+        original_stdout = sys.stdout
+        sys.stdout = output_captured
+        console = Console()
+        # show_locals set to False to limit tokens sent to OpenAI
+        console.print_exception(show_locals=False)
```

Comment:
Do we need to use console here? Not sure why its neccessary and if it is we should use it in the other standard hook as well i suppose?

Reply:
It's the rich output console which gives a much more detailed traceback. Currently the standard hook is printing just the exception type and not the traceback. Not sure if we'd want to output the full traceback? In any case, we only need the `rich` console if we're getting that full traceback.

Reply:
Id say we should be consistent, so if we do this here, then lets update the other one

Reply:
Ok so we'll have the full `rich` exception now instead of the standard Python one. Will make the change.

Reply:
Done